### PR TITLE
Closes #1530 Adding theme-debug tooling for lando and ddev.

### LINKED ISSUE #1530: Add `devel` and enable theme debugging to lando tooling
## Motivation
As a developer, often I have to enable theme debugging when creating quickstart pull-requests, and the current procedure, while in the docs is a pain.

## Proposed Resolution
Add lando tooling, so I don't have to remember how to do this, or look it up.

### Describe the solution you'd like
```
  # Enable theme debugging and devel.
  devel:
    service: appserver
    user: root
    cmd:
      - echo "Enabling theme debugging and devel..."
      - chmod 755 /app/web/sites/default/settings.php
      - /app/vendor/bin/drush --root=/app/web en -y devel
      - /app/vendor/bin/drupal --root=/app/web site:mode dev
      - chmod 644 /app/web/sites/default/settings.php
```

Then update the CONTRIBUTING docs to let people know about it.

### Describe alternatives you've considered
Leave everything as it is.

### PR DESCRIPTION
**Pull request by trackleft**
## Description

Adding tooling to easily enable theme debugging.

## Related issues
Closes #1530 
Decided on not including the enable devel module step, since that is easy enough to do with
`lando drush en -y devel`

## How to test
This is a developer convenience command when using lando to update theme files.

```
git clone https://github.com/az-digital/az_quickstart.git issues-1530
cd issues-1530
git checkout issues/1530
lando start
lando install
lando theme-debug

```
This is a developer convenience command when using ddev to update theme files.

```
git clone https://github.com/az-digital/az_quickstart.git issues-1530
cd issues-1530
git checkout issues/1530
ddev config --project-type php
ddev start
ddev install
ddev theme-debug

```

Theme debugging should successfully enable.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [x] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .lando.yml [hunk 1]
```diff
@@ -63,6 +63,16 @@ tooling:
   install:
     service: appserver
     cmd: /app/vendor/bin/drush --root=/app/web site:install --account-mail=noreply@email.arizona.edu --account-name=azadmin --account-pass=azadmin --db-url=mysql://drupal9:drupal9@database:3306/drupal9 -y --verbose
+  # Enable theme debugging and devel.
+  devel:
+    service: appserver
+    user: root
+    cmd:
+      - echo "Enabling theme debugging and devel..."
+      - chmod 755 -R /app/web/sites/default
```
Comment (trackleft) [inline .lando.yml:72]:
```suggestion
```

File: .lando.yml [hunk 2]
```diff
@@ -63,6 +63,16 @@ tooling:
...
+      - /app/vendor/bin/drush --root=/app/web en -y devel
+      - /app/vendor/bin/drupal --root=/app/web site:mode dev
+      - chmod 644 -R /app/web/sites/default
```
Comment (trackleft) [inline .lando.yml:75]:
```suggestion
```

File: .ddev/commands/web/devel [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+## Description: Enable theme debugging and devel module.
+## Usage: devel
+## Example: "ddev devel"
+
+echo "Enabling theme debugging and devel..."
```
Comment (trackleft) [inline .ddev/commands/web/devel:7]:
```suggestion
echo "Enabling theme debugging..."
```

File: .ddev/commands/web/devel [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+## Description: Enable theme debugging and devel module.
```
Comment (trackleft) [inline .ddev/commands/web/devel:3]:
```suggestion
## Description: Enable theme debugging.
```

File: .ddev/commands/web/devel [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+## Usage: devel
+## Example: "ddev devel"
```
Comment (trackleft) [inline .ddev/commands/web/devel:5]:
```suggestion
## Example: "ddev theme-debug"
```

File: .ddev/commands/web/devel [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+## Description: Enable theme debugging and devel module.
+## Usage: devel
```
Comment (trackleft) [inline .ddev/commands/web/devel:4]:
```suggestion
## Usage: theme-debug
```

File: .ddev/commands/web/devel [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+## Example: "ddev devel"
+
+echo "Enabling theme debugging and devel..."
+drush --root=/var/www/html/web en -y devel
```
Comment (trackleft) [inline .ddev/commands/web/devel:8]:
```suggestion
```

File: .lando.yml [hunk 8]
```diff
@@ -63,6 +63,14 @@ tooling:
...
     service: appserver
     cmd: /app/vendor/bin/drush --root=/app/web site:install --account-mail=noreply@email.arizona.edu --account-name=azadmin --account-pass=azadmin --db-url=mysql://drupal9:drupal9@database:3306/drupal9 -y --verbose
+  # Enable theme debugging and devel.
+  devel:
+    service: appserver
+    user: root
+    cmd:
+      - echo "Enabling theme debugging and devel..."
+      - /app/vendor/bin/drush --root=/app/web en -y devel
+      - /app/vendor/bin/drupal --root=/app/web site:mode dev
```
Comment (trackleft) [inline .lando.yml:73]:
```suggestion
  # Enable theme debugging and devel.
  devel:
    service: appserver
    user: root
    cmd:
      - echo "Enabling theme debugging..."
      - /app/vendor/bin/drupal --root=/app/web site:mode dev
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -270,19 +270,21 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
 
 ### Lando
-First you have to make the folder containing /app/web/sites/default/services.yml writable.
-```
-lando ssh --user root
-chmod 755 -R /app/web/sites/default
-exit
-```
-Then enable devel module and put the site into dev mode via drupal console.
+There is a provided lando tooling for enabling the `devel` module, and enabling
```
Comment (trackleft) [inline CONTRIBUTING.md:273]:
```suggestion
There is a provided lando tooling for enabling 
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -270,19 +270,21 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
+theme debugging.
```
Comment (trackleft) [inline CONTRIBUTING.md:274]:
```suggestion
There is a provided lando tooling for enabling theme debugging.
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -270,19 +270,21 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
+
+Simply use the following command once your site is installed:
+
 ```
-lando drush en -y devel
-lando drupal site:mode dev
+lando devel
```
Comment (trackleft) [inline CONTRIBUTING.md:279]:
```suggestion
lando theme-debug
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -270,19 +270,21 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
 ```
 
 ### DDev
+There is a provided ddev tooling for enabling the `devel` module, and enabling
+theme debugging.
+
+Simply use the following command once your site is installed:
+
```
Comment (trackleft) [inline CONTRIBUTING.md:287]:
```suggestion
There is a provided ddev tooling for enabling theme debugging.

Simply use the following command once your site is installed:

```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -270,19 +270,21 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
 ```
-ddev drush en -y devel && ddev drupal site:mode dev
+ddev devel
```
Comment (trackleft) [inline CONTRIBUTING.md:289]:
```suggestion
ddev theme-debug
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -270,19 +270,19 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
 
 ### Lando
-First you have to make the folder containing /app/web/sites/default/services.yml writable.
-```
-lando ssh --user root
-chmod 755 -R /app/web/sites/default
-exit
-```
-Then enable devel module and put the site into dev mode via drupal console.
+There is a provided lando tooling for enabling theme debugging.
```
Comment (trackleft) [inline CONTRIBUTING.md:273]:
```suggestion
There Lando tooling provided for enabling theme debugging.
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -270,19 +270,19 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
 
 ### Lando
-First you have to make the folder containing /app/web/sites/default/services.yml writable.
-```
-lando ssh --user root
-chmod 755 -R /app/web/sites/default
-exit
-```
-Then enable devel module and put the site into dev mode via drupal console.
+There Lando tooling provided for enabling theme debugging.
```
Comment (trackleft) [inline CONTRIBUTING.md:273]:
```suggestion
Quickstart provides Lando tooling for enabling theme debugging.
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -270,19 +270,19 @@ Developing within Drupal can be a real challenge without debugging enabled.
...
+Simply use the following command once your site is installed:
+
 ```
-lando drush en -y devel
-lando drupal site:mode dev
+lando theme-debug
 ```
 
 ### DDev
+There is a provided ddev tooling for enabling theme debugging.
```
Comment (trackleft) [inline CONTRIBUTING.md:282]:
```suggestion
Quickstart provides DDev tooling for enabling theme debugging.
```

**Comment by trackleft** [comment]
Alternatively to this PR, we could just set the drupal console tooling to `root` user.

**Review by akslay — DISMISSED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Comment by trackleft** [comment]
> During testing, I'm seeing the theme debug comments added to the rendered pages, but a cache clear is throwing up a white screen with

I've seen that error randomly with or without this PR

**Pull request merged**